### PR TITLE
feat(client): ts-migrate for files under client/src/component/search folder

### PR DESCRIPTION
**Pull request by krishna-saurav**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Below changes are implemented:
- Renamed the files to .tsx/.ts
- Added types/interface as required.
- Removed the Proptypes
- tested the code in gitpod
- Renamed the files in kebab case
- Formatted the files using prettier
- Resolved most of the Eslint issues.

### CODE REVIEW

**Comment by krishna-saurav** [comment]
@ojeytonwilliams  Please review.

**Comment by samarmohan** [comment]
Hey, can you fix the conflicts, please?

**Comment by krishna-saurav** [comment]
> Nice job(!), but can you fix the conflicts, please? Thank you!



> Nice job(!), but can you fix the conflicts, please? Thank you!

@samarmohan  done!!

**Review by raisedadead — COMMENTED** [review]
Hey there,

Thank you for taking the time to contribute to the TypeScript migration. 

The main issue has been updated a lot over the last few days, and so we suggest you re-read it and update this pull request to match if you have the time.

The main goals for a PR are:
- [ ] Changing the filename from `PascalCase` to `kebab-case` and ensuring that the imports in code are all still valid.
- [ ] Changing the file extension to .tsx (if a file has React JSX syntax)
- [ ] Fixing as many TypeScript and Eslint errors as you can while refactoring the code.

If you need any assistance, please reach out in our [contributing chat room](https://chat.freecodecamp.org/channel/contributors).

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Hello there. Thank you, for taking the time to make these changes.

I do not think we should be making any changes to `redux/` files, at this time. So, would you mind reverting the changes to those files?

**Comment by krishna-saurav** [comment]
@ShaunSHamilton  np at all. I'll revert the changes related to redux.

File: client/src/components/search/searchBar/search-bar.tsx [hunk 1]
```diff
@@ -176,9 +187,10 @@ export class SearchBar extends Component {
   render() {
     const { isDropdownEnabled, isSearchFocused, innerRef, t } = this.props;
     const { index } = this.state;
-    const placeholder = t('search.placeholder');
+    const placeholder = t ? t('search.placeholder') : '';
 
     return (
+<<<<<<< HEAD:client/src/components/search/searchBar/SearchBar.js
```
Comment (nhcarrigan) [inline client/src/components/search/searchBar/search-bar.tsx:193]: Merge conflict starts here

File: client/src/components/search/searchBar/search-bar.tsx [hunk 2]
```diff
@@ -212,13 +224,43 @@ export class SearchBar extends Component {
...
+                handleHits={this.handleHits}
+                handleMouseEnter={this.handleMouseEnter}
+                handleMouseLeave={this.handleMouseLeave}
+                selectedIndex={index}
+              />
+            )}
+          </div>
+        </HotKeys>
+      </div>
+>>>>>>> 918483ed0c (typescript migration for files under search folder):client/src/components/search/searchBar/search-bar.tsx
```
Comment (nhcarrigan) [inline client/src/components/search/searchBar/search-bar.tsx:258]: Merge conflict ends here.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Hey there,

Looks like a stray merge conflict made it into this PR.

File: client/src/components/search/searchBar/search-bar.tsx (hunk 1 above)
Comment (krishna-saurav) [reply client/src/components/search/searchBar/search-bar.tsx]: Yes. Resolved it.

**Pull request merged**